spec/requests/finance/lead_providers_controller_spec.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "Lead Providers for Finance users", type: :request do
+  let(:name) { Faker::Name.name }
+  let(:email) { Faker::Internet.email }
+  let(:finance_user) { create(:user, :finance) }
+  let!(:lead_providers) { create_list(:lead_provider, 5) }
+
+  before do
+    sign_in finance_user
+  end
+
+  describe "GET /finance/lead_providers" do
+    it "renders the index template" do
+      get "/finance/lead_providers"
+
+      expect(response).to render_template("finance/lead_providers/index")
+      expect(
+        Nokogiri.parse(response.body).css("table.govuk-table").text.gsub(/\n([ \t]*\n)+/, "\n").gsub(/^\s+|\s+$/, "").split("\n"),
+      ).to include(*lead_providers.map(&:name))
```

Comment:
Uhh, can we use `assigns` to check the lead providers, and maybe have a cypress tests to have visuals and accessibility check?